src/EventProducerService.cpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) Martin Da Costa 2023 (martindc.merg@gmail.com)
+// This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
+// Licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+// The full licence can be found at: http://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// TODO: Add AREQ and ASRQ opcode support.
+// Check error messages
+// Set Params flags
+
+#include <Streaming.h>
+#include "EventProducerService.h"
+#include <Controller.h>
+#include <vlcbdefs.hpp>
+
+namespace VLCB {
+
+void EventProducerService::setController(Controller *cntrl) 
+{
+  this->controller = cntrl;
+  this->module_config = cntrl->module_config;
+}
+
+void EventProducerService::begin()
+{
+  //Initialise instantMode
+  instantMode = module_config->currentMode;
+  //DEBUG_SERIAL << F("> instant MODE initialise as: ") << instantMode << endl;
+  
+  if (instantMode == MODE_UNINITIALISED)
+  {
+    return;
+  }
+  else
+  {
+    setProducedEvents();
+  }
+}
+
+void EventProducerService::setProducedEvents()
+{
+  if (module_config->EE_PROD_EVENTS > 0)
+  {
+    for (byte i = 0; i < module_config->EE_PROD_EVENTS; i++)
+    {
+      if (module_config->getEvTableEntry(i) == 0)
+      {
+        // Event does not exist so create it
+        byte data[4];
+        data[0] = highByte(module_config->nodeNum);
+        data[1] = lowByte(module_config->nodeNum);
+        data[2] = 0;
+        data[3] = i;
+        
+        module_config->writeEvent(i, data);  //index = i
+        module_config->writeEventEV(i, 1, 1);
+        module_config->writeEventEV(i, 2, 0);  //set default switch type
```

Comment:
Please add a comment to describe what EV1 and EV2 do.
What if the user sketch has set EE_NUM_EVS < 2 ?

Reply:
Comment added
There needs to be a trap for such an error.  Need to ad to the TODO list.